File: app/src/main/java/com/example/whattodo/ItemEvent_ComentFragment.java
package com.example.whattodo;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.example.whattodo.adapter.ComentAdapter;
import com.example.whattodo.adapter.EventAdapter;
import com.example.whattodo.model.Coment;
import com.example.whattodo.model.Event;
import com.firebase.ui.firestore.FirestoreRecyclerOptions;
import com.google.firebase.firestore.CollectionReference;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.Query;

import java.util.ArrayList;


public class ItemEvent_ComentFragment extends Fragment {

    View mView;
    private RecyclerView mRecyclerView;
    private ComentAdapter adapter;
    private FirebaseFirestore mFirestore;
    DocumentReference eventRef;
    String documentID;


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        mView = inflater.inflate(R.layout.fragment_eventitem_coment, container, false);

        documentID = getActivity().getIntent().getExtras().getString("DOCUMENT_KEY");
        mFirestore = FirebaseFirestore.getInstance();
        eventRef = mFirestore.collection("entryObject_DB").document(documentID);

        Query query = eventRef
                .collection("coment")
                .limit(10);

        initRecyclerView(query);

        return mView;
    }

    private void initRecyclerView(Query query) {

        FirestoreRecyclerOptions<Coment> options = new FirestoreRecyclerOptions.Builder<Coment>().setQuery(query, Coment.class).build();

        adapter = new ComentAdapter(options);

        mRecyclerView = mView.findViewById(R.id.coment_recycler_view);
        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        mRecyclerView.setAdapter(adapter);
    }

    @Override
    public void onStart() {
        super.onStart();
        adapter.startListening();
    }

    @Override
    public void onStop() {
        super.onStop();
        adapter.stopListening();
    }
}

File: app/src/main/java/com/example/whattodo/ItemEvent_ComentFragment.kt
package com.example.whattodo

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.whattodo.adapter.ComentAdapter
import com.example.whattodo.model.Coment
import com.firebase.ui.firestore.FirestoreRecyclerOptions
import com.google.firebase.firestore.DocumentReference
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.Query

class ItemEvent_ComentFragment : Fragment() {
    private lateinit var viewOfLayout: View
    private lateinit var mRecyclerView: RecyclerView

    private var adapter: ComentAdapter? = null
    private lateinit var mFirestore: FirebaseFirestore
    private lateinit var eventRef: DocumentReference

    private lateinit var b_afegircoment: Button

    var documentID: String? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        viewOfLayout = inflater.inflate(R.layout.fragment_eventitem_coment, container, false)

        mRecyclerView = viewOfLayout.findViewById(R.id.coment_recycler_view)
        mFirestore = FirebaseFirestore.getInstance()

        documentID = requireActivity().intent.extras!!.getString("DOCUMENT_KEY")

        eventRef = mFirestore.collection("entryObject_DB").document(documentID!!)
        val query = eventRef.collection("coment").limit(10)

        b_afegircoment = viewOfLayout.findViewById(R.id.b_comentari)
        b_afegircoment.setOnClickListener(View.OnClickListener {
            val intent = Intent(context, ActivityComent::class.java)
            intent.putExtra("DOCUMENT_KEY", documentID)
            startActivity(intent)
        })

        initRecyclerView(query)
        return viewOfLayout
    }

    private fun initRecyclerView(query: Query) {
        val options = FirestoreRecyclerOptions.Builder<Coment>().setQuery(query, Coment::class.java).build()
        mRecyclerView.setHasFixedSize(true)

        adapter = ComentAdapter(options)

        mRecyclerView.layoutManager = LinearLayoutManager(context)
        mRecyclerView.adapter = adapter
    }

    override fun onStart() {
        super.onStart()
        adapter!!.startListening()
    }

    override fun onStop() {
        super.onStop()
        adapter!!.stopListening()
    }
}